framework/source/class/qx/util/Function.js

```diff
@@ -52,9 +52,10 @@ qx.Bootstrap.define("qx.util.Function", {
     debounce: function (callback, delay, immediate) {
       var fired = false;
       var intervalId = null;
+      var args = null;
       var wrapperFunction = function () {
         // store the current arguments to have access inside the interval method
-        var args = qx.lang.Array.fromArguments(arguments);
+        args = qx.lang.Array.fromArguments(arguments);
```

Comment:
What was the incorrect behavior of the original code? The difference here is that args is now retaining closure on debounce instead of only wrapperFunction being retained, but `fromArguments` is called regardless, and nothing uses args outside of wrapperFunction. I'm not seeing a functional change here...???

Reply:
Take this example from the provided test case:
```
var debouncedTest = qx.util.Function.debounce(test, 10);
 debouncedTest(true);
 debouncedTest(false);
```
Without the fix the debounced function is called with argument `true`, with the fix it is called with argument `false`.

The wrong behaviour might have been introduced with this change: 9c972d955229e5e86c7928951557195058055013

bus i havent't checked that.

Reply:
Adding: That's because `args` is defined inside of the wrapper function and thus is persisted in the callback. There's no chance of getting the last value in there.

Reply:
Maybe it's just too early in the morning. I'm not seeing how it could possibly make a difference. debounce() returns the wrapper function. The first thing the wrapper function does is convert `arguments` to `args` as an array. That should be taking the arguments from the current call to debouncedTest, whether args is defined at line 58 or at line 55. The callback will use (and retain by clousre, if necessary) the current value of args, on each call to debouncedTest.

Since it makes a difference, I am clearly missing something. I don't see any harm in this PR, but would love to figure out why it should improve anything.

Reply:
If the `args` is defined inside the wrapper function, it has a different "pointer" every time the wrapper is called. The inner (setInterval) function only has the *initial* "pointer". Moving `args` one level up makes sure that the "pointer" is the same in every subsequent call.

Reply:
@peuter, yes -
 it was introduced by https://github.com/qooxdoo/qooxdoo/commit/9c972d955229e5e86c7928951557195058055013